features/nanostack/targets/TARGET_NXP/TARGET_KW41Z/NanostackRfPhyKw41z.cpp

```diff
@@ -0,0 +1,1093 @@
+    /* Mask SEQ, RX, TX and CCA interrupts */
+    ZLL->PHY_CTRL |= ZLL_PHY_CTRL_CCAMSK_MASK |
+                     ZLL_PHY_CTRL_RXMSK_MASK  |
+                     ZLL_PHY_CTRL_TXMSK_MASK  |
+                     ZLL_PHY_CTRL_SEQMSK_MASK;
+
+    while (ZLL->SEQ_STATE & ZLL_SEQ_STATE_SEQ_STATE_MASK) {}
+
+    irqStatus = ZLL->IRQSTS;
+    /* Mask TMR3 interrupt */
+    irqStatus |= ZLL_IRQSTS_TMR3MSK_MASK;
+    /* Clear transceiver interrupts except TMRxIRQ */
+    irqStatus &= ~(ZLL_IRQSTS_TMR1IRQ_MASK |
+                    ZLL_IRQSTS_TMR2IRQ_MASK |
+                    ZLL_IRQSTS_TMR3IRQ_MASK |
+                    ZLL_IRQSTS_TMR4IRQ_MASK);
+    ZLL->IRQSTS = irqStatus;
+}
+
+static void PhyIsrTimeoutCleanup(void)
+{
+    uint32_t irqStatus;
+
+    /* Set the PHY sequencer back to IDLE and disable TMR3 comparator and timeout */
+    ZLL->PHY_CTRL &= ~(ZLL_PHY_CTRL_TMR3CMP_EN_MASK |
+                       ZLL_PHY_CTRL_TC3TMOUT_MASK   |
+                       ZLL_PHY_CTRL_XCVSEQ_MASK);
+    /* Mask SEQ, RX, TX and CCA interrupts */
+    ZLL->PHY_CTRL |= ZLL_PHY_CTRL_CCAMSK_MASK |
+                     ZLL_PHY_CTRL_RXMSK_MASK  |
+                     ZLL_PHY_CTRL_TXMSK_MASK  |
+                     ZLL_PHY_CTRL_SEQMSK_MASK;
+
+    while (ZLL->SEQ_STATE & ZLL_SEQ_STATE_SEQ_STATE_MASK) {}
+
+    irqStatus = ZLL->IRQSTS;
+    /* Mask TMR3 interrupt */
+    irqStatus |= ZLL_IRQSTS_TMR3MSK_MASK;
+    /* Clear transceiver interrupts except TMR1IRQ and TMR4IRQ. */
+    irqStatus &= ~( ZLL_IRQSTS_TMR1IRQ_MASK |
+                    ZLL_IRQSTS_TMR4IRQ_MASK );
+    ZLL->IRQSTS = irqStatus;
+
+    /* The packet was transmitted successfully, but no ACK was received */
+    if (device_driver.phy_tx_done_cb) {
+        device_driver.phy_tx_done_cb(rf_radio_driver_id, rf_mac_handle, PHY_LINK_TX_SUCCESS, 1, 1);
+    }
+}
+
+/*
+ * \brief Function get the time-out for the current sequence.
+ *
+ * \return sequence time-out value
+ */
+static uint32_t rf_get_timeout(void)
+{
+    return ZLL->T3CMP;
+}
+
+/*
+ * \brief Function set a time-out to an XCVR sequence.
+ *
+ * \param pEndTime sequence time-out value [symbols]
+ *
+ * \return none
+ */
+static void rf_set_timeout(uint32_t pEndTime)
+{
+    uint32_t irqsts;
+
+    platform_enter_critical();
+
+    /* disable TMR3 compare */
+    ZLL->PHY_CTRL &= ~ZLL_PHY_CTRL_TMR3CMP_EN_MASK;
+    ZLL->T3CMP = pEndTime & ZLL_T3CMP_T3CMP_MASK;
+
+    /* acknowledge TMR3 IRQ */
+    irqsts  = ZLL->IRQSTS & BM_ZLL_IRQSTS_TMRxMSK;
+    irqsts |= ZLL_IRQSTS_TMR3IRQ_MASK;
+    ZLL->IRQSTS = irqsts;
+    /* enable TMR3 compare */
+    ZLL->PHY_CTRL |= ZLL_PHY_CTRL_TMR3CMP_EN_MASK;
+    /* enable autosequence stop by TC3 match */
+    ZLL->PHY_CTRL |= ZLL_PHY_CTRL_TC3TMOUT_MASK;
+
+    platform_exit_critical();
+
+}
+
+/**
+ * Call this only One time
+ */
+static void rf_mac_hw_init(void)
+{
+    xcvrStatus_t xcvrStatus;
+    uint32_t phyReg;
+
+    xcvrStatus = XCVR_Init(ZIGBEE_MODE, DR_500KBPS);
```

Comment:
Clarification needed here, what kind of configuration this actually creates. Thread uses 256kbps rate.

Reply:
This helps setup the hardware, below are the data rate options available:
https://github.com/NXPmicro/mbed/blob/Add_KW41Z_Thread/targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/middleware/wireless/TARGET_FRAMEWORK_5_3_3/XCVR/MKW41Z4/fsl_xcvr.h#L445

Reply:
For Thread the data rate should be 256kbps. I suppose ZIGBEE_MODE refers to 802.15.4 spec?
Are you planning to support other standards with this driver?